plugins/omdtls/omdtls.c

```diff
@@ -0,0 +1,885 @@
+		}
+	}
+finalize_it:
+	DBGPRINTF("dtls_send[%p]: Exit with %d for %s:%s\n", pWrkrData, iRet, pData->target, pData->port);
+	if(iRet != RS_RET_OK) {
+		dtls_close(pWrkrData);
+	}
+	RETiRet;
+
+	// Return sent
+	return lenSent;
+}
+
+/*
+*	Setup socket and establish virtual Connect on a SOCKET Level for DTLS 
+*/
+static rsRetVal
+dtls_connect(wrkrInstanceData_t *pWrkrData) {
+	DEFiRet;
+	int iErr;
+	instanceData *const pData = (instanceData *const) pWrkrData->pData;
+	BIO *bio_client;
+	const SSL_CIPHER *sslCipher;
+
+	DBGPRINTF("dtls_connect[%p]: setup DTLS for %s:%s\n", pWrkrData, pData->target, pData->port);
+	pWrkrData->bIsConnecting = 1;
+
+	// Create new SSL 
+	pWrkrData->sslClient = SSL_new(pData->pNetOssl->ctx);
+	if(!pWrkrData->sslClient) {
+		dbgprintf("dtls_connect[%p]: SSL_new failed failed\n", pWrkrData);
+		net_ossl_lastOpenSSLErrorMsg(pData->target, 0, pWrkrData->sslClient, LOG_WARNING, "omdtls", "SSL_new");
+		ABORT_FINALIZE(RS_RET_ERR);
+	}
+
+// MOVE TO SETUP
+	// Set certifcate check callback
+	if (pData->authMode != OSSL_AUTH_CERTANON) {
+		dbgprintf("dtls_connect[%p]: enable certificate checking (Mode=%d, VerifyDepth=%d)\n",
+			pWrkrData, pData->authMode, pData->CertVerifyDepth);
+		/* Enable certificate valid checking */
+		net_ossl_set_ssl_verify_callback(pWrkrData->sslClient, SSL_VERIFY_PEER|SSL_VERIFY_FAIL_IF_NO_PEER_CERT);
+		if (pData->CertVerifyDepth != 0) {
+			SSL_set_verify_depth(pWrkrData->sslClient, pData->CertVerifyDepth);
+		}
+	} else {
+		dbgprintf("dtls_connect[%p]: disable certificate checking\n", pWrkrData);
+		net_ossl_set_ssl_verify_callback(pWrkrData->sslClient, SSL_VERIFY_NONE);
+	}
+ 
+	/* Create BIO from socket array! */
+	bio_client = BIO_new_dgram(pWrkrData->sockout, BIO_NOCLOSE);
+	if (!bio_client) {
+		net_ossl_lastOpenSSLErrorMsg(pData->target, 0, pWrkrData->sslClient, LOG_INFO,
+			"dtls_connect", "BIO_new_dgram");
+		ABORT_FINALIZE(RS_RET_ERR);
+	}
+	BIO_ctrl(bio_client, BIO_CTRL_DGRAM_SET_CONNECTED, 0, &pWrkrData->dtls_client_addr);
+	SSL_set_bio(pWrkrData->sslClient, bio_client, bio_client);
+
+	/* Set debug Callback for conn BIO as well! */
+	net_ossl_set_bio_callback(bio_client);
+
+	/* Blocking socket */
+//	BIO_set_nbio( bio_client, 0 );
+//	SSL_set_bio(pWrkrData->sslClient, bio_client, bio_client);
+//	SSL_set_connect_state(pWrkrData->sslClient); /*sets ssl to work in client mode.*/
+
+	dbgprintf("dtls_connect[%p]: Starting DTLS session ...\n", pWrkrData);
+	/* Perform handshake */
+	iErr = SSL_connect(pWrkrData->sslClient);
+	if (iErr <= 0) {
+		net_ossl_lastOpenSSLErrorMsg(pData->target, iErr, pWrkrData->sslClient, LOG_INFO,
+			"dtls_connect", "SSL_connect");
+		ABORT_FINALIZE(RS_RET_ERR);
+	}
+
+	// Print Cipher info
+	sslCipher = SSL_get_current_cipher(pWrkrData->sslClient);
+	dbgprintf("dtls_connect[%p]: Cipher Version: %s Name: %s\n", pWrkrData,
+		SSL_CIPHER_get_version(sslCipher), SSL_CIPHER_get_name(sslCipher));
+/*
+	// Print Peer Certificate info
```

Comment:
remove commented-out code

Reply:
done, but not removed - useful for debugging

Reply:
shouldn't we then just guard it by an if and enable if debugging is turned on? Makes much more sense to me. Once rsyslog packages are deployed, you can no longer change compile time settings at all ;-)